pkg/collector/collector.go

```diff
@@ -0,0 +1,237 @@
+	protocol, addr, err := parseEndpointWithFallbackProtocol(endpoint, unixProtocol)
+	if err != nil {
+		return "", nil, err
+	}
+	if protocol != unixProtocol {
+		return "", nil, errOnlySupportUnixSocket
+	}
+
+	return addr, dial, nil
+}
+
+func dial(ctx context.Context, addr string) (net.Conn, error) {
+	return (&net.Dialer{}).DialContext(ctx, unixProtocol, addr)
+}
+
+func parseEndpointWithFallbackProtocol(endpoint string, fallbackProtocol string) (protocol string, addr string, err error) {
+	if protocol, addr, err = parseEndpoint(endpoint); err != nil && protocol == "" {
+		fallbackEndpoint := fallbackProtocol + "://" + endpoint
+		protocol, addr, err = parseEndpoint(fallbackEndpoint)
+		if err != nil {
+			return "", "", err
+		}
+	}
+	return protocol, addr, err
+}
+
+func parseEndpoint(endpoint string) (string, string, error) {
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return "", "", fmt.Errorf("error while parsing: %w", err)
+	}
+
+	switch u.Scheme {
+	case "tcp":
+		return "tcp", u.Host, nil
+	case "unix":
+		return "unix", u.Path, nil
+
+	case "":
+		return "", "", fmt.Errorf("using %q as %w", endpoint, errEndpointDeprecated)
+
+	default:
+		return u.Scheme, "", fmt.Errorf("%q: %w", u.Scheme, errProtocolNotSupported)
+	}
+}
+
+func getAllImages(c Client) ([]eraserv1alpha1.Image, error) {
+	backgroundContext, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	images, err := c.listImages(backgroundContext)
+	if err != nil {
+		return nil, err
+	}
+
+	allImages := make([]eraserv1alpha1.Image, 0, len(images))
+
+	for _, img := range images {
+		currImage := eraserv1alpha1.Image{
+			Digest: img.Id,
+		}
+		if len(img.RepoTags) > 0 {
+			currImage.Name = img.RepoTags[0]
+		}
+
+		allImages = append(allImages, currImage)
+	}
+
+	return allImages, nil
+}
+
+func createCollectorCR(ctx context.Context, allImages []eraserv1alpha1.Image) error {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Info("Could not create InClusterConfig")
+		return err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Info("Could not create clientset")
+		return err
+	}
+
+	imageCollector := eraserv1alpha1.ImageCollector{
+		TypeMeta: v1.TypeMeta{
+			APIVersion: "eraser.sh/v1alpha1",
+			Kind:       "ImageCollector",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			// imagejob will set node name as env when creating collector pod
+			Name: "imagecollector-" + os.Getenv("NODE_NAME"),
+		},
+		Spec: eraserv1alpha1.ImageCollectorSpec{
+			Images: allImages,
+		},
+	}
+
+	body, err := json.Marshal(imageCollector)
+	if err != nil {
+		log.Info("Could not marshal imagecollector for node: ", os.Getenv("NODE_NAME"))
+		return err
+	}
+
+	_, err = clientset.RESTClient().Post().
+		AbsPath(apiPath).
+		Resource("imagecollectors").
+		Body(body).DoRaw(ctx)
+
+	if err != nil {
+		log.Error(err, "ERROR: Could not create imagecollector", imageCollector.Name, imageCollector.APIVersion)
```

Comment:
```suggestion
		log.Error(err, "could not create imagecollector", imageCollector.Name, imageCollector.APIVersion)
```

no need to say error